inspire_hal/tasks.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 - 2019 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""HAL tasks."""
+
+from __future__ import absolute_import, division, print_function
+
+import datetime
+import os
+import socket
+
+import smtplib
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+
+from inspire_hal.bulk_push import run
+
+
+def hal_push(limit, yield_amt, mailing_list=None):
+    """Run a hal push."""
+
+    print('HAL: Starting to process HAL records')
+    if mailing_list:
+        send_hal_push_start_email(mailing_list)
+
+    total, now, ok, ko = run(
+        limit=limit,
+        yield_amt=yield_amt,
+    )
+
+    print(
+        'HAL: Finished, %s records processed in %s: %s ok, %s ko'
+        % (total, now, ok, ko)
+    )
+    if mailing_list:
+        send_hal_push_summary_email(
+            mailing_list=mailing_list,
+            total=total,
+            ok=ok,
+            now=now,
+            ko=ko,
+        )
+
+
+def send_hal_push_start_email(mailing_list):
+    host = socket.gethostname()
+    body = '''
+Hi!!
+
+Hal push just started to run in host %s, you should receive an email once it's
+finished too with the logs.
+
+Cheers!
+The Inspire developers team
+    ''' % (host)
+    message = MIMEText(body)
+    timestamp = datetime.datetime.now().strftime('[%Y/%m/%d-%H:%M:%S]')
+    message['Subject'] = 'Hal push starting at %s' % timestamp
+    message['From'] = 'inspire-halpush@cern.ch'
+    message['To'] = mailing_list
+
+    cli = smtplib.SMTP('localhost')
+    try:
+        cli.sendmail(
+            from_addr=message['From'],
+            to_addrs=[message['To']],
+            msg=message.as_string(),
+        )
+    finally:
+        cli.close()
+
+
+def send_hal_push_summary_email(mailing_list, total, ok, now, ko, attached_files=None):
```

Comment:
is the email send normally through the container? does it have SMTP etc?

Reply:
removed email and replaced by Zulip message